homeassistant/components/cover/brunt.py

```diff
@@ -0,0 +1,168 @@
+DEFAULT_NAME = 'brunt blind engine'
+NOTIFICATION_ID = 'brunt_notification'
+NOTIFICATION_TITLE = 'Brunt Cover Setup'
+
+CLOSED_POSITION = 0
+OPEN_POSITION = 100
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the brunt platform."""
+    # pylint: disable=no-name-in-module
+    from brunt import BruntAPI
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    bapi = BruntAPI(username=username, password=password)
+    try:
+        things = bapi.getThings()['things']
+        if not things:
+            raise HomeAssistantError
+
+        add_devices(BruntDevice(
+            hass, bapi, thing['NAME'],
+            thing['thingUri']) for thing in things)
+    except (TypeError, KeyError, NameError, ValueError) as ex:
+        _LOGGER.error("%s", ex)
+        hass.components.persistent_notification.create(
+            'Error: {}<br />'
+            'You will need to restart hass after fixing.'
+            ''.format(ex),
+            title=NOTIFICATION_TITLE,
+            notification_id=NOTIFICATION_ID)
+
+
+class BruntDevice(CoverDevice):
+    """Representation of a Brunt cover device.
+
+    Contains the common logic for all Brunt devices.
+    """
+
+    def __init__(self, hass, bapi, name, thing_uri):
+        """Init the Brunt device."""
+        self._bapi = bapi
+        self._name = name
+        self._thing_uri = thing_uri
+
+        self._state = None
+        self._available = None
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the device as reported by tellcore."""
+        return self._name
+
+    @property
+    def available(self):
+        """Could the device be accessed during the last update call."""
+        return self._available
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        data = {}
+        if self._state['moveState'] == 1:
+            data[ATTR_COVER_STATE] = 'OPENING'
+        elif self._state['moveState'] == 2:
+            data[ATTR_COVER_STATE] = 'CLOSING'
+        elif int(self._state['currentPosition']) == CLOSED_POSITION:
+            data[ATTR_COVER_STATE] = 'CLOSED'
+        elif int(self._state['currentPosition']) == OPEN_POSITION:
+            data[ATTR_COVER_STATE] = 'OPENED'
+        else:
+            data[ATTR_COVER_STATE] = 'PARTIALLY OPENED'
+        data[ATTR_CURRENT_POSITION] = int(self._state['currentPosition'])
+        data[ATTR_REQUEST_POSITION] = int(self._state['requestPosition'])
+        return data
+
+    @property
+    def state(self):
+        """Return the state of the cover."""
+        return STATE_CLOSED if int(
+            self._state['currentPosition']) == CLOSED_POSITION else STATE_OPEN
```

Comment:
Looking at device_state_attributes, you can improve the states (add STATE_OPENING, STATE_CLOSING)

Reply:
Hi @dgomes thanks for the review, I've made a bunch of changes, I've now gotten rid of the ATTR_COVER_STATE as a device attribute, and added opening and closing states (as well as my own partially opened but not moving state). I just need to test it and then I will commit a new version.

Reply:
Don't add states that don't exist in the frontend code. They will simply not show